lib/services/api/request_editor.rb

```diff
@@ -0,0 +1,14 @@
+module Api
+  class RequestEditor
+    def self.edit(request, data)
+      request_options = RequestParser.parse_options(data)
+      user = RequestParser.parse_user(data) || User.current_user
+
+      begin
+        request.update_request(request_options, user)
+      rescue => err
+        raise BadRequestError, "Could not update the request - #{err}"
```

Comment:
I see the error message would change for the ones currently implementing edit, one way this can be solved is by sending a request type:

```
RequestEditor.edit(request, request_type, data)
```

and leverage in line 10, maintaining current behavior.